spec/api/v1/dropdown_data_spec.rb

```diff
@@ -119,34 +119,87 @@
 
-    context 'filtering' do
-      context 'by rep order date' do
-        context 'using a scheme 9 date' do
+    context 'when filtering' do
+      context 'with rep order and main hearing dates' do
+        context 'with no dates' do
           it 'defaults to scheme 9 offences' do
             is_expected.to match_array([exposed_offence[scheme_9_offence]])
           end
+        end
+
+        context 'with a scheme 9 rep order data and no main hearing date' do
+          before { params[:rep_order_date] = '2016-03-01' }
 
           it 'returns scheme 9 offences' do
-            params[:rep_order_date] = '2016-03-01'
             is_expected.to match_array([exposed_offence[scheme_9_offence]])
           end
         end
 
-        context 'using a scheme 10 date' do
+        context 'with a scheme 10 rep order data and no main hearing date' do
+          before { params[:rep_order_date] = '2018-04-01' }
+
           it 'returns scheme 10 offences' do
-            params[:rep_order_date] = '2018-04-01'
             is_expected.to match_array([exposed_offence[scheme_10_offence]])
           end
         end
 
-        context 'using a scheme 11 date' do
+        context 'with a scheme 11 rep order data and no main hearing date' do
+          before { params[:rep_order_date] = '2018-12-31' }
+
           it 'returns scheme 11 offences' do
-            params[:rep_order_date] = '2018-12-31'
             is_expected.to match_array([exposed_offence[scheme_11_offence]])
           end
         end
+
+        context 'with a scheme 12 rep order data and no main hearing date' do
+          before { params[:rep_order_date] = '2020-09-17' }
+
+          it 'returns scheme 12 offences' do
+            is_expected.to match_array([exposed_offence[scheme_12_offence]])
+          end
+        end
+
+        context 'with a scheme 13 rep order data and no main hearing date' do
+          before { params[:rep_order_date] = '2022-09-30' }
+
+          it 'returns scheme 13 offences' do
+            is_expected.to match_array([exposed_offence[scheme_13_offence]])
+          end
+        end
+
+        context 'with a rep order between 17/09/22 and 20/09/22; no main hearing date' do
+          before { params[:rep_order_date] = '2022-09-29' }
+
+          it 'returns scheme 12 offences' do
+            is_expected.to match_array([exposed_offence[scheme_12_offence]])
+          end
+        end
+
+        context 'with a rep order between 17/09/20 and 20/09/22; main hearing date before 31/10/22' do
+          before do
+            params[:rep_order_date] = '2022-09-29'
+            params[:main_hearing_date] = '2022-10-30'
+          end
+
+          it 'returns scheme 12 offences' do
+            is_expected.to match_array([exposed_offence[scheme_12_offence]])
+          end
+        end
+
+        context 'with a rep order between 17/09/20 and 20/09/22; main hearing date on 31/10/22' do
```

Comment:
all lgtm, i have a nitpick but it's not a blocker

could it be named 'on or after' e.g. `main hearing date on or after 31/10/22`

or maybe have a new context to test when it's after the 31st e.g. 1/11/22